action.yml

```diff
@@ -34,7 +34,24 @@ runs:
         repo: MarcoIeni/release-plz
         tag: ${{ inputs.version }}
         cache: enable
-    - run: |
+    - id: commit-author
+      name: Determine Git user info from GitHub token
+      shell: bash
+      run: |
+        VIEWER_JSON=$(gh api graphql -f query='query { viewer { name login databaseId }}' --jq '.data.viewer')
+        VIEWER_NAME=$(jq --raw-output '.name | values' <<< "${VIEWER_JSON}")
+        VIEWER_LOGIN=$(jq --raw-output '.login' <<< "${VIEWER_JSON}")
+        VIEWER_DATABASE_ID=$(jq --raw-output '.databaseId' <<< "${VIEWER_JSON}")
+
+        USER_NAME="${VIEWER_NAME:-${VIEWER_LOGIN}}"
+        USER_EMAIL="${VIEWER_DATABASE_ID}+${VIEWER_LOGIN}@users.noreply.github.com"
+
+        echo "name=${USER_NAME}" >> "${GITHUB_OUTPUT}"
+        echo "email=${USER_EMAIL}" >> "${GITHUB_OUTPUT}"
+      # env GITHUB_TOKEN from action
```

Comment:
what's the purpose of this comment? Is the GITHUB_TOKEN env variable used in `gh` here or we need to do something else?

Reply:
my concern is that if we don't use a token for the `gh` cli we could get rate-limited. 🤔

Reply:
[Here](https://cli.github.com/manual/gh_auth_login#:~:text=Alternatively%2C%20gh%20will%20use%20the,token%20%7D%7D%20to%20%22env%22.) it says that we can set the `GH_TOKEN` env variable. Should we run gh as `GH_TOKEN=GITHUB_TOKEN gh api ...`?

Reply:
The comment just says that the `GITHUB_TOKEN` environment variable is implicit and used/required here, just like in the other tasks in the action further below it.
IMHO the better design would be that instead of:
```yml
      - name: Run release-plz
        uses: MarcoIeni/release-plz-action@v0.5
        with: # <--- Input variables
          command: release-pr
          registry: my-registry
          project_manifest: rust-crates/my-crate/Cargo.toml
          version: release-plz-v0.2.45
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
To use an explicit input variable for the token:
```yml
inputs:
  token:
    description: "INPUT: GitHub token with access as per https://release-plz.ieni.dev/docs/github/trigger "
    required: false
    default: ${{ github.token }}
```
```yml
      - name: Run release-plz
        uses: MarcoIeni/release-plz-action@v0.5
        with: # <--- Input variables
          command: release-pr
          registry: my-registry
          project_manifest: rust-crates/my-crate/Cargo.toml
          version: release-plz-v0.2.45
          token: ${{ secrets.MY_PAT }}
```
```yml
- run: |
    …
  env:
    GITHUB_TOKEN: ${{ inputs.token }}
```
But that would be an unrelated change for this pull request.

The `gh` CLI needs a token either as `GH_TOKEN` / `GITHUB_TOKENM` env var (as per the documentation you linked or https://cli.github.com/manual/gh_help_environment ). We need it to use the token we use in our Action, because we want to determine the "viewer" data from that very token (i.e. default / PAT / App). It is unlikely we get rate limited since we always use a token per design here.
GitHub Actions always at least have the default token available as `GITHUB_TOKEN` env or `${{ github.token }}` variable ( https://docs.github.com/en/actions/learn-github-actions/contexts#github-context ).

Reply:
Or in other words, it says that the following is implicit:
```yml
- run: |
    …
  env:
    GITHUB_TOKEN: ${{ env.GITHUB_TOKEN }}
```
because it is redundant.

Reply:
@MarcoIeni any updates on this?

I can create a new pull request for the Action to take the token as an input and then rebase this one here onto it once you've merged the other one.

Would you like me to change the code or explain something?

I could add tests for the PAT and App cases, but those would require you to add repository secrets.

Reply:
Hey, sorry, but I didn't have the time to look into this yet. I will try to have a look in the following days 👍